homeassistant/components/plex/server.py

```diff
@@ -39,12 +42,37 @@
 plexapi.X_PLEX_VERSION = X_PLEX_VERSION
 
 
+def debounce(func):
+    """Decorate function to debounce callbacks from Plex websocket."""
+
+    unsub = None
+
+    async def call_later_listener(self, _):
+        """Handle call_later callback."""
+        nonlocal unsub
+        unsub = None
+        await self.hass.async_add_executor_job(func, self)
+
+    @wraps(func)
+    def wrapper(self):
+        """Schedule async callback."""
+        nonlocal unsub
+        if unsub:
+            _LOGGER.debug("Throttling update of %s", self.friendly_name)
+            unsub()  # pylint: disable=not-callable
```

Comment:
unsub functions should only be called from the same context as in which they are set. So if created with `async_call_later`, need to be called from inside the event loop. 

Also, because this is running in a thread, you cannot guarantee that nothing else touches it. You should instead run this all inside the event loop.

Reply:
It's better to turn this the `wrapper` function into an async function.

Reply:
Really sorry about that, hopefully fixed in #33730.